MapApp/Controllers/LocationsController.cs

```diff
@@ -32,8 +32,20 @@ public async Task<IActionResult> Index()
 			return View(await _context.Location.ToListAsync());
 		}
 
-		// GET: Locations
-		public async Task<IActionResult> Data()
+        public async Task<IActionResult> NameFilter(string name)
+        {
+            if (!String.IsNullOrEmpty(name))
+            {
+                return View("Index", (await _context.Location.ToListAsync()).Where(s => s.Name.Contains(name)));
```

Comment:
Consider using Contains case insensitive (https://stackoverflow.com/questions/17563929/how-to-make-string-contains-case-insensitive)
```
 s.Name.Contains(name, , StringComparison.OrdinalIgnoreCase)